client/yelp/yelp.go

```diff
@@ -0,0 +1,75 @@
+package yelp
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+const searchBusinessesEndpoint = "https://api.yelp.com/v3/businesses/search"
+
+func NewYelpClient(apiKey string) YelpClient {
+	return YelpClient{
+		apiKey: apiKey,
+		searchBusinessesEndpoint: searchBusinessesEndpoint,
```

Comment:
we should make this Endpoint URL more generic. Perhaps make this a base url equal to `https://api.yelp.com/` and then the function `SearchBusinesses` for example could append `v3/businesses/search`. And other functions that hit other endpoints could do the same. Similar to this http://hassansin.github.io/Unit-Testing-http-client-in-Go

Reply:
Great suggestion will add.